src/test/java/testresults/PostTestResultsRoadworthiness.java

```diff
@@ -0,0 +1,171 @@
+        testResultsSteps.statusCodeShouldBe(200);
+
+        // Verify TestNumber/CertificateNumber are  test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+    @Title("CVSB-7675 - TC3 - Negative (HGV) - FAIL")
+    @Test
+    public void testResults_Roadworthiness_HGV_Fail_Certificate_Number() {
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_roadworthiness_hgv_fail_7675.json", "$");
+
+        // Create alteration to add one more tech record to in the request body
+        String randomVin = GenericData.generateRandomVin();
+        String randomTestResultId = UUID.randomUUID().toString();
+        JsonPathAlteration alterationVin = new JsonPathAlteration("$.vin", randomVin, "", "REPLACE");
+        JsonPathAlteration alterationTestResultId = new JsonPathAlteration("$.testResultId", randomTestResultId, "", "REPLACE");
+        JsonPathAlteration alterationResult = new JsonPathAlteration("$.testTypes[0].testResult", "fail", "", "REPLACE");
+
+        // Collate the list of alterations.
+        List<JsonPathAlteration> alterations = new ArrayList<>(Arrays.asList(
+                alterationVin,
+                alterationResult,
+                alterationTestResultId));
+
+        // Post the results, together with any alterations, and verify that they are accepted.
+        testResultsSteps.postVehicleTestResultsWithAlterations(testResultRecord, alterations);
+        testResultsSteps.statusCodeShouldBe(201);
+        testResultsSteps.validateData("Test records created");
+
+        // Retrieve the created record, and verify that the fields are present.
+        testResultsSteps.getTestResults(randomVin);
+        testResultsSteps.statusCodeShouldBe(200);
+
+        // Verify CertificateNumber field have the null value
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", null);
+    }
+
+    @Title("CVSB-7675 - TC4 - Negative (TRL) - FAIL")
+    @Test
+    public void testResults_Roadworthiness_TRL_Fail_Certificate_Number() {
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_roadworthiness_trl_fail_7675.json", "$");
+
+        // Create alteration to add one more tech record to in the request body
+        String randomVin = GenericData.generateRandomVin();
+        String randomTestResultId = UUID.randomUUID().toString();
+        JsonPathAlteration alterationVin = new JsonPathAlteration("$.vin", randomVin, "", "REPLACE");
+        JsonPathAlteration alterationTestResultId = new JsonPathAlteration("$.testResultId", randomTestResultId, "", "REPLACE");
+        JsonPathAlteration alterationResult = new JsonPathAlteration("$.testTypes[0].testResult", "fail", "", "REPLACE");
```

Comment:
this is unnecessary since the json file already has "testResult" set as "fail"

Reply:
Changes Done.